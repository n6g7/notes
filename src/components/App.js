import React, { PureComponent } from "react"
import { Switch, Route } from "react-router-dom"

import { Notebook, Summary } from "@pages"

class App extends PureComponent {
  render() {
    return (
      <Switch>
        <Route exact path="/" component={Summary} />
        <Route path="/nb/:notebook" component={Notebook} />
      </Switch>
    )
  }
}

export default App
