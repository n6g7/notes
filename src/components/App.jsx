import React, { PureComponent } from 'react';
import { Link } from 'react-router';

import githubIcon from '../images/github.svg';
import notebooks from '../../notebooks';

class App extends PureComponent
{
  render() {
    return <main>
      <header>
        <h1>notes</h1>
        <nav>
          <ul>
            { Object.keys(notebooks).map(name =>
              <li key={`nb-${name}`}>
                <Link to={`/nb/${name}`}>{name}</Link>
              </li>
            )}
          </ul>
          <aside>
            <a href="https://github.com/n6g7/notes" title="GitHub repository" target="blank">
              <img src={githubIcon} alt="GitHub" />
            </a>
          </aside>
        </nav>
      </header>
      <article className="container">
        {this.props.children}
      </article>
    </main>;
  }
}

export default App;
