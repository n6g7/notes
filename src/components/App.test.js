import React from "react"
import { shallow } from "enzyme"

import App from "./App"

describe("<App />", () => {
  it("renders well", () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
