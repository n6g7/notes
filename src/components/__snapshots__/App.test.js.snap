// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders well 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/nb/:notebook"
  />
</Switch>
`;
